---
title: "PlateletHW_Data Report"
subtitle: "A Study of Platelet Data"
output:
  pdf_document: default
  html_document: default
date: "2024-11-09"
author: "Buranee Manoapan"
student_id: "6737932"
---

One of the main causes of death and disability in the globe is coronary artery disease (CAD). Atherosclerosis-induced coronary artery constriction is the hallmark of CAD, a progressive illness that frequently leads to potentially fatal outcomes like myocardial infarction and stroke. Antiplatelet therapy is a crucial component of managing CAD, and medications such as clopidogrel are essential for lowering the risk of thrombosis.The platelet aggregation rate (PAR), a gauge of platelet responsiveness to agonists like adenosine diphosphate (ADP), is a crucial biomarker for evaluating the efficacy of antiplatelet medication. In a cohort of 211 CAD patients, this study aimed to examine the connection between PAR, genetic polymorphisms, and clopidogrel resistance. It specifically targeted three genetic variations: PON1.192Q\>R (rs662), CYP2C19\**2 (rs4244285), and CYP2C19\**3 (rs4986893). These genes participate in enzymatic processes that control clopidogrel bioactivation and platelet activity. In order to clarify the effect of these polymorphisms on PAR and clopidogrel resistance, the study combined statistical and association analysis. The dataset contained genotypes for the three target single nucleotide polymorphisms (SNPs), ADP-induced platelet aggregation levels, and demographic factors (age and sex).*R software* (version 2024.04.2+764) was aims to analyze and visualize the statistical analysis by plotting. Firstly, The data was import and read by *read_tsv function* and checking for an outliner by using IQR rule base on *quantile function.* Absolute and logarithmic transformations were used to guarantee data normalisation, producing theoretical, non-negative values. After cleaning, the data was stored for later examinations. To investigate the connection, an association analysis were perform by linear regression, multiple linear regression, and logistic regression. The result from linear regression show statistically significant associations with ADP levels of rs4244285 and rs4986893 which can be observed through the expected value of the dependent variable, whereas rs662 didn't indicate any association with ADP levels. Multiple linear regression model highlights the impact of rs4244285 and rs4986893 on ADP levels as result shown as estimate, p-value., and multiple R-squared. Finally, logistic regression also confirmed that both SNPs located within CYP2C19 gene are strong predictors of resistance. The model supported that r662 nor does, age, and sex are non-significant predictors.

# Import Raw Data

```{r}
library(readr)
PlateletHW <- read_tsv("raw_data/PlateletHW.tsv")
PlateletHW
dim(PlateletHW)
```

# Check for Outliners

IQR rules indicate that ADP value aren't identified as outliners.

```{r}
quantiles <- quantile(PlateletHW$ADP, probs = c(0.25, 0.75))
iqr <- quantiles[2] - quantiles[1]
upper_limit <- quantiles[2] + 1.5 * iqr
lower_limit <- quantiles[1] - 1.5 * iqr
outliers <- PlateletHW$ADP[PlateletHW$ADP < lower_limit | PlateletHW$ADP > upper_limit]
print(outliers)
```

# Clean data

```{r}
library(dplyr)
PlateletHW_clean <- PlateletHW %>%filter(ADP >= lower_limit & ADP <= upper_limit)
summary(PlateletHW_clean$ADP)
write_tsv(PlateletHW_clean,"clean_data/PlateletHW_clean.tsv")
head(PlateletHW_clean)

```

# Data Normalization

To normalize the ADP taking log transformation to get the positive integer and avoiding 0 and save the file.

```{r}
PlateletHW$ADP[PlateletHW$ADP == 0] <- mean(PlateletHW$ADP[PlateletHW$ADP > 0], na.rm = TRUE)
PlateletHW <- PlateletHW %>% mutate(ADP_log = log(ADP + 1))
PlateletHW$ADP_log[is.na(PlateletHW$ADP_log)] <- mean(PlateletHW$ADP_log, na.rm = TRUE)
PlateletHW$ADP <- PlateletHW$ADP_log
PlateletHW$ADP_log <- NULL
PlateletHW <- PlateletHW[, c("IID", "ADP", setdiff(names(PlateletHW_clean), c("IID", "ADP")))]
summary(PlateletHW$ADP)
dim(PlateletHW)

```

# Analyses the data

## Test for Associations: Linear regression model for each SNP

The association of each SNP has tested to observe the relationship between each SNP and ADP levels, a statistically significant of rs4244285 (figure 1.1), rs4986893 (figure 1.2), and rs662 (figure 1.3) can be viusulize through the plot of linear regression.

```{r}
model_rs4244285 <- lm(ADP ~ rs4244285, data = PlateletHW)
model_rs4986893 <- lm(ADP ~ rs4986893, data = PlateletHW)
model_rs662 <- lm(ADP ~ rs662, data = PlateletHW)

summary(model_rs4244285)
summary(model_rs4986893)
summary(model_rs662)
```

The summarize of model in each SNPs can be visualize through ggplot.

```{r}
library(ggplot2)
ggplot(PlateletHW, aes(x = rs4244285, y = ADP)) + geom_point() + geom_smooth(method = "lm", se = FALSE, color = "red") + labs(title = "Figure 1.1 Linear Regression of ADP on rs4244285")
```

```{r}
ggplot(PlateletHW, aes(x = rs4986893, y = ADP)) +geom_point() + geom_smooth(method = "lm", se = FALSE, color = "red") + labs(title = "Figure 1.2 Linear Regression of ADP on rs4986893")
```

```{r}
ggplot(PlateletHW, aes(x = rs662, y = ADP)) +geom_point() + geom_smooth(method = "lm", se = FALSE, color = "red") + labs(title = "Figure 1.3 Linear Regression of ADP on rs662")
```

## Multiple Linear Regression

The relationship between the dependent variable as `ADP` and independent variable including `rs4244285`, `rs4986893`, `rs662`, `AGE`, and `SEX` are show the significant predictors, insignificant predictors, and model fit.

```{r}
model <- lm(ADP ~ rs4244285 + rs4986893 + rs662 + AGE + SEX, data = PlateletHW)
summary(model)
```

## Test an association for Clopidogrel Resistance (Binary Outcome)

The performance of logistic regression model with all SNPs, age, and sex is use to confirm the relationship of the factors that influence ADP of the dataset. The model interpret that significant predictors are rs4244285 (CYP2C192) and rs4986893 (CYP2C193) due to p-value\<0.05 whereas rs662 (PON1.192Q\>R) is not show strong association with CR.

```{r}
resistance_model <- glm(Resistance ~ rs4244285 + rs4986893 + rs662 + AGE + SEX, data = PlateletHW, family = binomial)
summary(resistance_model)
```

Note: The confusion matrix is perform to gather an understanding on the model that classify resistance case.

```{r}
predicted_probs <- predict(model, type = "response")
predicted_classes <- ifelse(predicted_probs > 0.5, 1, 0)  
actual_classes <- PlateletHW$Resistance
confusion_matrix <- table(Predicted = predicted_classes, Actual = actual_classes)
print(confusion_matrix)
```
